SITL logger compile error with GCC 7.3

**Bug**
I updated Cygwin to the newest version and it's using GCC 7.3

```
$ gcc --version
gcc (GCC) 7.3.0
```

When I compile SITL by `make posix` I get the following errors which weren't there before on this branch:

```
../../src/modules/logger/logger.cpp:2136:27: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
statfs_buf.f_bavail >= (px4_statfs_buf_f_bavail_t)(min_free_bytes / statfs_buf.f_bsize)) {
~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../../src/modules/logger/logger.cpp:2172:26: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
if (statfs_buf.f_bavail < (px4_statfs_buf_f_bavail_t)(50 * 1024 * 1024 / statfs_buf.f_bsize)) {
~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Workaround**
Of course I can just add a `(int)` cast in front of the second operand of the comparison and then it compiles again but I'm writing this issue because there's likely some inconsistency with the types.

Why is `px4_statfs_buf_f_bavail_t` defined once signed, once usigned here? https://github.com/PX4/Firmware/blob/9ce83f22087311df06078edf3a9da640d93ab396/src/platforms/px4_defines.h#L115-L173

--- a/src/modules/logger/logger.cpp
+++ b/src/modules/logger/logger.cpp
@@ -64,11 +64,19 @@
 #include <replay/definitions.hpp>
 #include <version/version.h>
 
-#ifdef __PX4_DARWIN
+#if defined(__PX4_DARWIN)
 #include <sys/param.h>
 #include <sys/mount.h>
 #else
 #include <sys/statfs.h>
+#endif
+
+#if defined(__PX4_NUTTX)
+#define px4_statfs_buf_f_bavail_t int
+#elif defined(__PX4_DARWIN) || defined(__PX4_QURT)
+#define px4_statfs_buf_f_bavail_t unsigned long
+#else
+typedef decltype(statfs::f_bavail) px4_statfs_buf_f_bavail_t;
 #endif
 
 #define GPS_EPOCH_SECS ((time_t)1234567890ULL)

--- a/src/modules/logger/logger.cpp
+++ b/src/modules/logger/logger.cpp
@@ -71,13 +71,7 @@
 #include <sys/statfs.h>
 #endif
 
-#if defined(__PX4_NUTTX)
-#define px4_statfs_buf_f_bavail_t int
-#elif defined(__PX4_DARWIN) || defined(__PX4_QURT)
-#define px4_statfs_buf_f_bavail_t unsigned long
-#else
 typedef decltype(statfs::f_bavail) px4_statfs_buf_f_bavail_t;
-#endif
 
 #define GPS_EPOCH_SECS ((time_t)1234567890ULL)
 

--- a/src/platforms/px4_defines.h
+++ b/src/platforms/px4_defines.h
@@ -170,7 +170,10 @@ using ::isfinite;
 #define USEC_PER_TICK (1000000/PX4_TICKS_PER_SEC)
 #define USEC2TICK(x) (((x)+(USEC_PER_TICK/2))/USEC_PER_TICK)
 
-#define px4_statfs_buf_f_bavail_t unsigned long
+#ifdef __cplusplus
+#include <sys/statfs.h>
+typedef decltype(statfs::f_bavail) px4_statfs_buf_f_bavail_t;
+#endif
 
 #ifdef __PX4_QURT
 

--- a/src/platforms/px4_defines.h
+++ b/src/platforms/px4_defines.h
@@ -112,7 +112,6 @@ typedef param_t px4_param_t;
 
 #define PX4_ROOTFSDIR ""
 #define _PX4_IOC(x,y) _IOC(x,y)
-#define px4_statfs_buf_f_bavail_t int
 
 // mode for open with O_CREAT
 #define PX4_O_MODE_777 0777
@@ -169,11 +168,6 @@ using ::isfinite;
 
 #define USEC_PER_TICK (1000000/PX4_TICKS_PER_SEC)
 #define USEC2TICK(x) (((x)+(USEC_PER_TICK/2))/USEC_PER_TICK)
-
-#ifdef __cplusplus
-#include <sys/statfs.h>
-typedef decltype(statfs::f_bavail) px4_statfs_buf_f_bavail_t;
-#endif
 
 #ifdef __PX4_QURT
 

